### PR TITLE
AList.init, AList.range, ASet.range

### PR DESCRIPTION
**Pull request by dsyme**

### CODE REVIEW

**Comment by dsyme** [comment]
Range was harder than I thought, is there an easier way? I was surprised I needed to keep the index list in the reader - necessary just to keep track of the indexes

**Comment by dsyme** [comment]
Am very much relying on the systematic testing here to cover all cases: https://github.com/fsprojects/FSharp.Data.Adaptive/pull/51/files#diff-5d4931e252ca277e0c22783029e31483R749

**Comment by dsyme** [comment]
Is it possible this should be `ASet.range: ... -> aset<int>` instead?  Then the indexes don't matter and don't need to be tracked?

**Comment by dsyme** [comment]
>Is it possible this should be ASet.range: ... -> aset<int> instead? Then the indexes don't matter and don't need to be tracked?

However I guess there's no easy way from `ASet<int>` (ordered, but ASet has no knowledge of this) to `AList<int>` (ordered)

**Comment by krauthaufen** [comment]
Hey, actually your implementation looks pretty much the way I imagined it to look like.
Maybe the range code could be simplified a little but I think it's good...

Yes, the aset range operation should be much easier but semantically both things kind of make sense to me.

**Comment by dsyme** [comment]
> Maybe the range code could be simplified a little but I think it's good...

Yes working on this :)

**Comment by dsyme** [comment]
OK, this is ready now.  I made the `range` a bit simpler sharing the code and made it generic on any integer type that has the right set of operations

File: src/FSharp.Data.Adaptive/AdaptiveIndexList/AdaptiveIndexList.fsi [hunk 1]
```diff
@@ -51,6 +51,22 @@ module AList =
     /// Creates an alist using the given reader-creator.
     val ofReader : create : (unit -> #IOpReader<IndexListDelta<'T>>) -> alist<'T>
     
+    /// Creates an alist using the given value generator
+    val constant: value: (unit -> IndexList<'T>) -> alist<'T>
```
Comment (dsyme) [inline src/FSharp.Data.Adaptive/AdaptiveIndexList/AdaptiveIndexList.fsi]: I made this public so we can inline creations of `AList` (the inlined implementation uses it) - is the name ok?  It's perhaps a bit counterintuitive - maybe `AList.constantDelayed` or `AList.ofIndexListDelayed`

**Comment by dsyme** [comment]
Hmmm no CI running ?

**Comment by krauthaufen** [comment]
Ah, sorry i messed up the CI for PRs obviously

**Comment by krauthaufen** [comment]
should be working as soon as you push something new to it I think

File: src/FSharp.Data.Adaptive/AdaptiveIndexList/AdaptiveIndexList.fsi (hunk 1 above)
Comment (krauthaufen) [reply src/FSharp.Data.Adaptive/AdaptiveIndexList/AdaptiveIndexList.fsi]: I think the name is just okay, the type and the comment make it clear what happens...

**Comment by krauthaufen** [comment]
nice 😀

**Comment by dsyme** [comment]
Thank you!

**Comment by krauthaufen** [comment]
Thank you 😁

**Comment by dsyme** [comment]
Hey when you get a chance can you push a new nuget?  

Also could you give my nuget account (dsyme) co-ownership on https://www.nuget.org/packages/FSharp.Data.Adaptive/?  Thanks

**Comment by dsyme** [comment]
> Also could you give my nuget account (dsyme) co-ownership on https://www.nuget.org/packages/FSharp.Data.Adaptive/? Thanks

May as well add me to Fable.React.Adaptive and Fable.Elmish.Adaptive too, though I'm not really maintaining those it at least gives them co-owners

Also if you get a chance, please unlist FsIncremental

**Comment by krauthaufen** [comment]
sent you lots of invites...
Hope it works (it sometimes doesn't)

**Comment by dsyme** [comment]
Thanks, got them and accepted them

**Pull request merged**